src/annotator/util/listener-collection.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @typedef Listener
+ * @prop {Window|Element|Document} eventTarget
+ * @prop {string} eventType
+ * @prop {(event: Event) => void} listener
+ */
+
+/**
+ * Utility that prevents forgetting to remove event listener.
+ */
+export function listenerCollection() {
+  /** @type{Listener[]} */
+  let listeners = [];
+
+  /**
+   * @param {Listener['eventTarget']} eventTarget
+   * @param {Listener['eventType']} eventType
+   * @param {Listener['listener']} listener
+   * @param {boolean | AddEventListenerOptions | undefined} [options]
+   */
+  function register(eventTarget, eventType, listener, options) {
```

Comment:
I wonder about naming these `add`/`removeAll` to be more similar to `addEventListener` and `removeEventListener`. Thoughts?

Reply:
No issues.